sekoia_automation/scripts/files_generator.py

```diff
@@ -174,8 +175,64 @@ def update_module_manifest(self, module: type[Module]):
             manifest = json.load(f)
 
         manifest["configuration"] = configuration_model.schema()
+        self.add_secrets_to_configuration(
+            manifest["configuration"], configuration_model.__fields__.values()
+        )
 
         with self.manifest_path.open("w") as out:
             out.write(json.dumps(manifest, indent=2))
 
         print(f"[green][+][/green] Updated {self.manifest_path}")
+
+    @staticmethod
+    def add_secrets_to_configuration(config, config_fields):
```

Comment:
I don't think relying on `alias` is the best way to achieve it.

It messes with the property names and has an other meaning in Pydantic.

We could rely on `extra` to do it:

```python
class ModuleConfiguration(BaseModel):
    module_field: str = Field(secret=True)

for field in ModuleConfiguration.__fields__.values():
    print(field.field_info.extra)  # {"secret": True}
```

Reply:
Also for now it seems the python code doesn't compute the `required` attribute. It would be awesome to compute it for all the required fields, not only secrets

Reply:
Check for all required fields: done 

Use field_info.extra: looking into it, you have a part of the doc referencing this? I couldn't find it

Reply:
I didn't see it in the doc, but `Field` has `**extra: Any` in its signature and pass it to a `FieldInfo` class that is then returned.

Reply:
Should be good now